app/CPReflectionService/RBObject+CocoaPods.m

```diff
@@ -52,18 +52,27 @@
 NSError * _Nonnull
 CPErrorFromException(NSException * _Nonnull exception, NSString * _Nullable message)
 {
-  CPErrorDomainCode code  = -1;
-  NSString *exceptionName = nil;
-  NSString *description   = nil;
-  NSArray *rubyBacktrace  = nil;
+  CPErrorDomainCode code = -1;
+
+  NSMutableDictionary *userInfo = [NSMutableDictionary new];
+  userInfo[CPErrorObjCBacktrace] = exception.callStackSymbols;
+  if (message) {
+    userInfo[NSLocalizedRecoverySuggestionErrorKey] = message;
+  }
 
   if ([exception.name hasPrefix:@"RBException_"]) {
     VALUE rb_exception = [exception.userInfo[@"$!"] __rbobj__];
 
     VALUE exceptionNameValue = rb_class_name(rb_obj_class(rb_exception));
-    exceptionName = @(StringValuePtr(exceptionNameValue));
-    description = @"Uncaught Ruby exception.";
-    rubyBacktrace = exception.userInfo[CPErrorRubyBacktrace];
+    userInfo[CPErrorName] = @(StringValuePtr(exceptionNameValue));
+    VALUE rb_cause = rb_funcall(rb_exception, rb_intern("cause"), 0);
```

Comment:
No that didn’t work well. It would crash when trying to serialise an error in `-description` and possibly also when sending over the wire.